src/Npgsql/NpgsqlBinaryExporter.cs

```diff
@@ -61,10 +61,8 @@ internal NpgsqlBinaryExporter(NpgsqlConnector connector)
     {
         _connector = connector;
         _buf = connector.ReadBuffer;
-        _typeMapper = connector.TypeMapper;
-        _columnLen = int.MinValue;   // Mark that the (first) column length hasn't been read yet
-        _column = -1;
-        _typeHandlerCache = null!;
+        _column = -2;
```

Comment:
_column now has 3 states:
- -2 before row
- -1 before first column
- \>= 0 on column

Reply:
Maybe define constants for -2 and -1 to make this slightly clearer?

Reply:
Yeah that'd be a bit nicer.